redfish-core/lib/hypervisor_system.hpp

```diff
@@ -255,6 +296,36 @@ inline bool extractHypervisorInterfaceData(
                                 {
                                     ipv4Address.origin = "DHCP";
                                 }
+
+                                if (!translateDHCPEnabledToBool(*dhcpEnabled,
+                                                                false))
+                                {
+                                    ipv6Address.origin = "Static";
+                                }
+                                else
+                                {
+                                    ipv6Address.origin = "DHCP";
+                                }
+                            }
+                        }
+                        else if (property.first == "DefaultGateway6")
+                        {
+                            const std::string* defaultGateway6 =
+                                std::get_if<std::string>(&property.second);
+                            if (defaultGateway6 != nullptr)
```

Comment:
I see (if defaultGateway6 == nullptr) { return internal error } for new code

Reply:
Done